Jenkinsfile

```diff
@@ -1,28 +1,110 @@
 #!groovy
 library "atomSharedLibraries@master"
 
+import static Constants.*
+
+class Constants {
+    static final mainDirectoryName = "contrail_state_files"
+    static final terraformStateFileName = "state"
+}
+
+def warningEcho(message) {
+    echo "\033[1;33m${message}\033[0m"
+}
+
+def ensureMainDirectoryExists() {
+    dirExistingResult = checkIfDirectoryExists("../${mainDirectoryName}")
+    if(dirExistingResult == '0') {
+        sh "mkdir ../${mainDirectoryName}"
+    }
+}
+
+def ensureUserDirectoryExists() {
+    dirExistingResult = checkIfDirectoryExists("../${mainDirectoryName}/${params.Login}")
+    if(dirExistingResult == '0') {
+        sh "mkdir ../${mainDirectoryName}/${params.Login}"
+    }
+}
+
+def checkIfDirectoryExists(name) {
+    return sh(script: "test -d \"${name}\" && echo '1' || echo '0' ", returnStdout: true).trim()
+}
+
+def checkIfFileExists(name) {
+    return sh(script: "test -f \"${name}\" && echo '1' || echo '0' ", returnStdout: true).trim()
+}
+
+def copyTerraformFiles() {
+    sh "cp provision/${params.orchestrator}/${params.orchestrator}.tf ../${mainDirectoryName}/${params.Login}/${params.MachineName}/${params.orchestrator}.tf"
+    sh "cp provision/${params.orchestrator}/variables.tf ../${mainDirectoryName}/${params.Login}/${params.MachineName}/variables.tf"
+}
+
+def copyDaemonFile() {
+    sh "cp provision/daemon.json ../${mainDirectoryName}/${params.Login}/${params.MachineName}/daemon.json"
```

Comment:
We're using global state in all these helper functions. It would be more maintainable if they were as pure as possible. I'll talk to you IRL.

Reply:
Done.